test/test_dht.cpp

```diff
@@ -186,15 +186,78 @@ void send_simple_dht_request(node& node, char const* msg, udp::endpoint const& e
 	}
 }
 
+void write_peers(entry::dictionary_type& r, std::set<tcp::endpoint> const& peers)
+{
+	entry::list_type& pe = r["values"].list();
+	for (std::set<tcp::endpoint>::const_iterator it = peers.begin()
+		 ; it != peers.end(); ++it)
+	{
+		std::string endpoint(18, '\0');
```

Comment:
I realize this code is basically unchanged, but it would be possible to call write_endpoint(*it, std::back_inserter(endpoint)) instead of this pre-allocation and resizing business, right?